community/bolt/src/main/java/org/neo4j/bolt/messaging/KnownType.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.bolt.messaging;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum KnownType
```

Comment:
I intentionally did not version this enum as there's a need for looking up all server supported types independent of its version. The version specific logic will still live in `PackerV1` and `PackerV2`.

Yep, that's right. I spent some time around removing those constants but removing them will either require an invocation of `KnownType#signature()` or a lookup through `KnownType#valueOf()` - which would cause the packing / unpacking logic to involve more code when compared to using constants. WDYT?

Reply:
The main concern for me is that we defined this magic `N`, `R` etc. twice.
How about `NODE( NeoPackV1.Node, "Node" )`? Will this be simple and clear to read?

Reply:
👍I think so.